txdbus/authentication.py

```diff
@@ -168,7 +176,7 @@ def _authGetDBusCookie(self, cookie_context, cookie_id):
         if dstat.st_uid != pwd.getpwuid(os.geteuid()).pw_uid:
             raise Exception('Keyrings directory is not owned by the current user. Aborting authentication!')
         
-        f = open(os.path.join(cookie_dir, cookie_context), 'r')
+        f = open(os.path.join(cookie_dir, cookie_context.decode('ascii')), 'rb')
```

Comment:
BTW: You could probably use `with os.path.join(cookie_dir, cookie_context.decode('ascii')), 'rb') as f:` here instead to make this a lot more readable